components/concept/engine/src/main/java/mozilla/components/concept/engine/webextension/WebExtension.kt

```diff
@@ -180,6 +181,13 @@ abstract class WebExtension(
      * available icon to the provided size.
      */
     abstract suspend fun loadIcon(size: Int): Bitmap?
+
+    /**
+     * An error that occur trying to perform an action like installing, removing or updating
+     * an [WebExtension] that is not recoverable.
+     */
+    class UnRecoverableError(throwable: Throwable) : Error(throwable)
```

Comment:
I wouldn't nest this under `WebExtension`. It can be in the same file, but also a top-level type.

Reply:
An alternative could be to have a top-level `GeckoWebExtensionUpdateError` or `GeckoWebExtensionInstallError` which wraps/contains the underlying exception. Then we can add a `isRecoverable` function to it directly?

Reply:
I thought that Errors were meant to express a [situation where an app should not try to recover from](https://docs.oracle.com/javase/7/docs/api/java/lang/Error.html), if that is correct the `Error` post-fix implies that the situation is not recoverable, for using an `isRecoverable` property  then we would  need an more generic exception `WebExtensionException` that could be useful for both cases recoverable and not recoverable. I omitted `Gecko` prefix as these type of exceptions can be handle in higher level component that is not aware of the Gecko or other engine implementation.

Reply:
`WebExtensionException` is a more generic as allows to cover installation/update

Reply:
Right, yeah, which is why I am suggesting we have a `WebExtension[Install|Update]Exception` with a `isRecoverable` function, instead of a type for a special case.

Reply:
The `Gecko` prefix you will need as the exception will wrap Gecko's, similar to how `GeckoWebExtension` is a `WebExtension` and wraps a native GeckoView extension.

Reply:
Higher up in the stack you would handle `WebExtensionException` (or `WebExtensionInstallException` if you want to call it that).

Reply:
> Right, yeah, which is why I am suggesting we have a WebExtension[Install|Update]Exception with a isRecoverable function, instead of a type for a special case.
👍🏽